R/hypothesize.R

```diff
@@ -17,71 +24,43 @@
 #'   generate(reps = 100, type = "permute") %>%
 #'   calculate(stat = "F")
 #'
+#' @importFrom purrr compact
 #' @export
-hypothesize <- function(x, null, ...) {
-  hypothesize_checks(x, null)
+hypothesize <- function(x, null, p = NULL, mu = NULL, med = NULL, sigma = NULL) {
 
+  # Custom logic, because using match.arg() would give a default value when
+  # the user didn't specify anything.
+  null <- match_null_hypothesis(null)
   attr(x, "null") <- null
 
-  dots <- list(...)
-
-  if ((null == "point") && (length(dots) == 0)) {
-    stop_glue(
-      "Provide a parameter and a value to check such as `mu = 30` for the ",
-      "point hypothesis."
-    )
-  }
-
-  if ((null == "independence") && (length(dots) > 0)) {
-    warning_glue(
-      "Parameter values are not specified when testing that two variables are ",
-      "independent."
-    )
-  }
-
-  if ((length(dots) > 0) && (null == "point")) {
-    params <- parse_params(dots, x)
-    attr(x, "params") <- params
-
-    if (any(grepl("p.", attr(attr(x, "params"), "names")))) {
-     # simulate instead of bootstrap based on the value of `p` provided
-      attr(x, "type") <- "simulate"
-    } else {
-      attr(x, "type") <- "bootstrap"
-    }
+  hypothesize_checks(x, null)
 
-  }
+  dots <- compact(list(p = p, mu = mu, med = med, sigma = sigma))
```

Comment:
Snazzy!

Reply:
That's all Richie!

Reply:
Thanks, @richierocks!

Reply:
Thanks for flagging that bit about plot labels. We'll try to tack that on to the next wave of updates, led by @simonpcouch, focused on documentation.

Reply:
Awesome! Looking forward to it. Happy to provide a quick review. Just tag me and I'll let you know a timeline for when I'll be able to get it done.

Reply:
I'll move it over to an issue and try to assign it to Simon.

Reply:
Great. I'm not sure why the travis build is taking so long...normally it's like a 5 min job.